frontend/src/state/models/schedule-data/shift-info.model.ts

```diff
@@ -1,5 +1,18 @@
 export type Shift = "R" | "P" | "D" | "N" | "DN" | "PN" | "RPN" | "W" | "U" | "L4" | null;
 
+export enum ShiftCode {
```

Comment:
Chyba nie da się zrobić inaczej, ponieważ typ - to faktycznie tak jak mówiłeś - alias dla TypeScript'a, który służy do tego, żeby w czasie kompilacji mieć pewność w tym, że ty dostajesz te wartości których oczekujesz.

Ale typ nie jest uwzględniany podczas runtime'u, ponieważ nie jest kompilowany.

 
Chyba że to jest klasa. Klasy mogą być normalnie przekonwertowane do javascriptu, i mogą być użyte do filtrowania 

![image](https://user-images.githubusercontent.com/21079319/93491405-efb72300-f909-11ea-812c-3d755b991938.png)


Btw, interfejsy też nie są kopilowane
![image](https://user-images.githubusercontent.com/21079319/93491509-0eb5b500-f90a-11ea-81cf-4cec18d9fa20.png)

Reply:
Oki to zostawię to tak jak jest. Dzięki za objaśnienie :D

Reply:
Nie ma za co :)